pkg/services/authn/authnimpl/sync/user_sync.go

```diff
@@ -305,14 +312,33 @@ func (s *UserSync) updateUserAttributes(ctx context.Context, usr *user.User, id
 		needsUpdate = true
 	}
 
+	span.SetAttributes(
+		attribute.String("identity.ID", id.ID),
+		attribute.String("identity.ExternalUID", id.ExternalUID),
+	)
+	if usr.IsProvisioned {
+		s.log.Debug("User is provisioned", "id,UID", id.UID)
+		needsConnectionCreation = false
+		authInfo, err := s.authInfoService.GetAuthInfo(ctx, &login.GetAuthInfoQuery{UserId: usr.ID, AuthModule: id.AuthenticatedBy})
+		if err != nil {
+			s.log.Error("Error getting auth info", "error", err)
+			return err
+		}
+
+		if id.ExternalUID != authInfo.ExternalUID {
```

Comment:
I think the `ExternalUID` should be stored encrypted or hashed. Wdyt?

Reply:
Yes! I thought about improving it in another PR. Thanks for agreeing with me. I'll do it in this PR.

Reply:
externalUID hashed as suggested

Reply:
One other thing: `authInfo.ExternalUID` shouldn't be empty in this case.

Reply:
changed as requested